pkg/internal/kube/informer_provider.go

```diff
@@ -168,27 +159,19 @@ func (mp *MetadataProvider) CurrentNodeName(ctx context.Context) (string, error)
 // for getting informer data
 func (mp *MetadataProvider) initLocalInformers(ctx context.Context) (*meta.Informers, error) {
 	done := make(chan error)
-	var informers *meta.Informers
-	go func() {
-		var err error
-		opts := append(disabledInformerOpts(mp.cfg.DisabledInformers),
-			meta.WithResyncPeriod(mp.cfg.ResyncPeriod),
-			meta.WithKubeConfigPath(mp.cfg.KubeConfigPath))
-		if informers, err = meta.InitInformers(ctx, opts...); err != nil {
-			done <- err
-		}
-		close(done)
-	}()
-
-	select {
-	case <-time.After(mp.cfg.SyncTimeout):
-		klog().Warn("kubernetes cache has not been synced after timeout. The kubernetes attributes might be incomplete."+
-			" Consider increasing the BEYLA_KUBE_INFORMERS_SYNC_TIMEOUT value", "timeout", mp.cfg.SyncTimeout)
-	case err, ok := <-done:
-		if ok {
-			return nil, fmt.Errorf("failed to initialize Kubernetes informers: %w", err)
-		}
```

Comment:
we didn't realize a bug here.

If the timeout happened, the later returned "informers" variable would be nil, as it requires that the `InitInformers` function successfully returns from another goroutine.